agent/src/main/java/reactor/blockhound/InstrumentationUtils.java

```diff
@@ -35,19 +42,59 @@ static void injectBootstrapClasses(Instrumentation instrumentation, String... cl
         try (ZipOutputStream zipOutputStream = new ZipOutputStream(new FileOutputStream(tempJarFile))) {
             for (String className : classNames) {
                 String classFile = className.replace(".", "/") + ".class";
-                InputStream inputStream = classLoader.getResourceAsStream(classFile);
-                ZipEntry e = new ZipEntry(classFile);
-                zipOutputStream.putNextEntry(e);
-
-                byte[] buf = new byte[4096];
-                int n;
-                while ((n = inputStream.read(buf)) > 0) {
-                    zipOutputStream.write(buf, 0, n);
+                try (InputStream inputStream = classLoader.getResourceAsStream(classFile)) {
```

Comment:
changes in this class don't look like a javadoc change to me :suspect:

Reply:
The thing is... I wanted to "hide" this class from public API (even tho it must be public to work), and "public API" was part of the scope of this PR.

I can submit it separately if you wish! Especially given that it is a separate commit :)

Reply:
https://github.com/reactor/BlockHound/pull/53

after another thought, submitted separately :)